contracts/core/EntryPoint.sol

```diff
@@ -261,7 +263,10 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard,
         unchecked {
             // handleOps was called with gas limit too low. abort entire bundle.
             if (
-                gasleft() < callGasLimit + mUserOp.verificationGasLimit + 5000
+                gasleft() <
+                callGasLimit +
+                mUserOp.paymasterPostOpGasLimit +
```

Comment:
At this point we already have a `context`, so if it is empty the `postOp` will not be called. I think it may make sense to allow the transaction to go on in this case, instead of always adding `paymasterPostOpGasLimit` to the required gas left?
Say, the UserOp have requested 1'000'000 gas for a `postOp` but it does not need it any more, why should this check fail?

Reply:
why should we (entrypoint) waste gas to check it? it is the paymaster's job to put a valid postOpGasLimit, even if context is empty.
There is another edge case: `validatePaymasterUserOp() returns a `context`, and the `postOpGasLimit` is zero: in this case, we still execute the callData and revert the postOp (on OOG) - again, because it is not the EntryPoint's job to validate the userop fields.